app/Decorators/calculator_decorator.rb

```diff
@@ -0,0 +1,51 @@
+class CalculatorDecorator
+  def initialize(result)
+    @result = result
+  end
+
+  def for_json
+    {
+      result: get_values,
+      date: get_date,
+      text_products_to_be_used: get_text_products_to_be_used,
+      text_products_used: get_text_products_used
+    }
+  end
+
+  private
+
+  def get_values
+    {
+      money_spent: @result.used_diapers_price,
+      money_will_be_spent: @result.to_be_used_diapers_price,
```

Comment:
create `attr_reader :result` and then use it anywhere (`result` instead of `@result`)